From: "Dr. Tobias Quathamer" <toddy@debian.org>
Date: Mon, 9 Feb 2026 20:40:50 +0100
Subject: Skip TestTSAN test on s390x.

The test consistently fails, and is probably a real bug.
It's currently being investigated, but until that is
resolved, we disable the test to be able to build the
golang package, so it can migrate to testing and
fix lots of other bugs and CVEs.

Forwarded: https://github.com/golang/go/issues/77289
---
 src/cmd/cgo/internal/testsanitizers/tsan_test.go | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/cmd/cgo/internal/testsanitizers/tsan_test.go b/src/cmd/cgo/internal/testsanitizers/tsan_test.go
index 58a1058..3628ee4 100644
--- a/src/cmd/cgo/internal/testsanitizers/tsan_test.go
+++ b/src/cmd/cgo/internal/testsanitizers/tsan_test.go
@@ -32,7 +32,9 @@ func TestTSAN(t *testing.T) {
 	if goarch == "riscv64" {
 		t.Skipf("skipping TestTSAN test on %s (see https://github.com/golang/go/issues/75553)", goarch)
 	}
-
+	if goarch == "s390x" {
+		t.Skipf("skipping TestTSAN test on %s (see https://github.com/golang/go/issues/77289)", goarch)
+	}
 
 	t.Parallel()
 	requireOvercommit(t)
